packages/kbn-i18n/src/core/i18n.test.js

```diff
@@ -0,0 +1,737 @@
+        month: 'short',
+        day: 'numeric',
+        year: 'numeric',
+      });
+
+      i18n.setFormats({
+        date: {
+          short: {
+            month: 'short',
+            day: 'numeric',
+            year: 'numeric',
+          },
+        },
+      });
+
+      expect(i18n.getFormats().date.short).toEqual({
+        month: 'short',
+        day: 'numeric',
+        year: 'numeric',
+      });
+
+      i18n.setFormats({
+        date: {
+          short: {
+            month: 'long',
+            day: 'numeric',
+            year: 'numeric',
+          },
+        },
+      });
+
+      expect(i18n.getFormats().date.short).toEqual({
+        month: 'long',
+        day: 'numeric',
+        year: 'numeric',
+      });
+    });
+  });
+
+  describe('getFormats', () => {
+    test('should return "en" formats by default', () => {
+      const { formats } = require('./formats');
+
+      expect(i18n.getFormats()).toEqual(formats);
+    });
+
+    test('should return updated formats', () => {
+      const { formats } = require('./formats');
+
+      i18n.setFormats({
+        foo: 'bar',
+      });
+
+      expect(i18n.getFormats()).toEqual({
+        ...formats,
+        foo: 'bar',
+      });
+    });
+  });
+
+  describe('getRegisteredLocales', () => {
+    test('should return empty array by default', () => {
+      expect(i18n.getRegisteredLocales()).toEqual([]);
+    });
+
+    test('should return array of registered locales', () => {
+      i18n.addMessages({
+        locale: 'en',
+      });
+
+      expect(i18n.getRegisteredLocales()).toEqual(['en']);
+
+      i18n.addMessages({
+        locale: 'ru',
+      });
+
+      expect(i18n.getRegisteredLocales()).toContain('en', 'ru');
+      expect(i18n.getRegisteredLocales().length).toBe(2);
+
+      i18n.addMessages({
+        locale: 'fr',
+      });
+
+      expect(i18n.getRegisteredLocales()).toContain('en', 'ru', 'fr');
+      expect(i18n.getRegisteredLocales().length).toBe(3);
+    });
+  });
+
+  describe('translate', () => {
+    test('should throw error if id is not a string', () => {
+      expect(() => i18n.translate(undefined)).toThrow();
+      expect(() => i18n.translate(null)).toThrow();
+      expect(() => i18n.translate(true)).toThrow();
+      expect(() => i18n.translate(5)).toThrow();
+      expect(() => i18n.translate({})).toThrow();
+    });
+
+    test('should throw error if id is empty string', () => {
+      expect(() => i18n.translate('')).toThrow();
+    });
+
+    test('should throw error if translation message and defaultMessage are not provided', () => {
+      expect(() => i18n.translate('foo')).toThrow();
+    });
+
+    test('should return message as is if values are not provided', () => {
+      i18n.init({
+        locale: 'en',
+        ['a.b.c']: 'foo',
+      });
+
+      expect(i18n.translate('a.b.c')).toBe('foo');
+    });
+
+    test('should return default message as is if values are not provided', () => {
```

Comment:
__suggestion:__ we should also have test when `defaultMessage` is provided __and__ non-default translation is registered for the particular id.